libbpf-rs/tests/test.rs

```diff
@@ -1793,3 +1793,28 @@ fn test_sudo_map_autocreate_disable() {
 
     open_obj.load().expect("failed to load object");
 }
+
+/// Check that we are able to attach using ksyscall
+#[test]
+fn test_sudo_attach_ksyscall() {
+    bump_rlimit_mlock();
+
+    let mut obj = get_test_object("ksyscall.bpf.o");
+    let prog = obj
+        .prog_mut("handle__ksyscall")
+        .expect("Failed to find program");
+
+    let _link = prog
+        .attach_ksyscall(false, "kill")
+        .expect("Failed to attach prog");
+
+    let map = obj.map("ringbuf").expect("Failed to get ringbuf map");
+    let action = || {
+        // This is fine as we won't be having a negative pid
+        let _ = std::process::Command::new("kill -9 -1000").spawn();
```

Comment:
Mind that negative PIDs have special meaning with kill.
man 2 kill
> If pid is less than -1, then sig is sent to every process in the process group whose ID is -pid

`kill`, the command line, seems to only care about -1 though.